src/sources/vector/mod.rs

```diff
@@ -117,17 +136,28 @@ pub struct VectorConfig {
     #[configurable(derived)]
     #[serde(default, deserialize_with = "bool_or_struct")]
     acknowledgements: SourceAcknowledgementsConfig,
+
+    /// The namespace to use for logs. This overrides the global setting.
+    #[serde(default)]
+    #[configurable(metadata(docs::hidden))]
+    log_namespace: Option<bool>,
 }
 
-impl GenerateConfig for VectorConfig {
-    fn generate_config() -> toml::Value {
-        toml::Value::try_from(Self {
+impl Default for VectorConfig {
+    fn default() -> Self {
+        Self {
             version: None,
             address: "0.0.0.0:6000".parse().unwrap(),
             tls: None,
             acknowledgements: Default::default(),
-        })
-        .unwrap()
+            log_namespace: None,
```

Comment:
Shower thought and not really that important, but I'm thinking this might be nicer as, especially now it's a part of `Default::default()`:

```rust
Self {
  address: "0.0.0.0:6000".parse().unwrap(),
  ..Default::default()
}

```

Reply:
I like the spirit of it but, it fails to compile due to recursive calls (which kinda makes sense)

```
function cannot return without recursing
`#[deny(unconditional_recursion)]` implied by `#[deny(warnings)]`
```

Reply:
Oh, yeah of course..